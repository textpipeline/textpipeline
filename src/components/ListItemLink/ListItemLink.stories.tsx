import List from '@material-ui/core/List';
import TwitterIcon from '@material-ui/icons/Twitter';
import { storiesOf } from '@storybook/react';
import React from 'react';

import ListItemLink from './ListItemLink';

storiesOf('ListItemLink', module).add('default', () => (
  <List>
    <ListItemLink text="link 1" href="#" />
    <ListItemLink text="link 2" href="#" />
    <ListItemLink text="external link 1" href="https://google.com" isExternal={true} />
    <ListItemLink text="external link 2" href="https://bbc.com" isExternal={true} />
    <ListItemLink text="link with icon" href="#" icon={TwitterIcon} />
  </List>
));
