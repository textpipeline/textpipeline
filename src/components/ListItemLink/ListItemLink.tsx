import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import SvgIcon from '@material-ui/core/SvgIcon';
import ArrowRightIcon from '@material-ui/icons/ArrowRight';
import React from 'react';

import ExternalLink from '../ExternalLink';
import InternalLink from '../InternalLink';

export interface ListItemLinkProps {
  readonly text: string;
  readonly href: string;
  readonly icon?: typeof SvgIcon;
  readonly isExternal?: boolean;
  readonly subtitle?: string;
}

const linkColor = 'textPrimary';

const ListItemLink: React.FC<ListItemLinkProps> = ({ icon, text, href, isExternal, subtitle }) => {
  const Icon = icon || ArrowRightIcon;
  const LinkComponent = isExternal ? ExternalLink : InternalLink;
  const inset = !Icon;

  return (
    <ListItem button component={LinkComponent} href={href} color={linkColor}>
      {Icon && (
        <ListItemIcon>
          <Icon />
        </ListItemIcon>
      )}
      <ListItemText inset={inset} primary={text} secondary={subtitle} />
    </ListItem>
  );
};

export default ListItemLink;
