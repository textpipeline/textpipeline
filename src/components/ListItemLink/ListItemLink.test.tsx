import TwitterIcon from '@material-ui/icons/Twitter';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import ListItemLink from './ListItemLink';

describe('ListItemLink', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    rendered = shallow(<ListItemLink text="Testing" href="https://example.com/foo" />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });

  describe('with an icon specified', () => {
    beforeEach(() => {
      rendered = shallow(<ListItemLink icon={TwitterIcon} text="Testing" href="https://example.com/foo" />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('with isExternal specified true', () => {
    beforeEach(() => {
      rendered = shallow(<ListItemLink text="Testing" href="https://example.com/foo" isExternal={true} />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('with a subtitle specified', () => {
    beforeEach(() => {
      rendered = shallow(<ListItemLink text="Testing" subtitle="Testing Subtitle" href="https://example.com/foo" />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });
});
