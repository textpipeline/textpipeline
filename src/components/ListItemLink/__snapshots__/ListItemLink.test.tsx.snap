// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemLink renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  color="textPrimary"
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  href="https://example.com/foo"
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <ArrowRightIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    inset={false}
    primary="Testing"
  />
</WithStyles(ForwardRef(ListItem))>
`;

exports[`ListItemLink with a subtitle specified renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  color="textPrimary"
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  href="https://example.com/foo"
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <ArrowRightIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    inset={false}
    primary="Testing"
    secondary="Testing Subtitle"
  />
</WithStyles(ForwardRef(ListItem))>
`;

exports[`ListItemLink with an icon specified renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  color="textPrimary"
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  href="https://example.com/foo"
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <TwitterIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    inset={false}
    primary="Testing"
  />
</WithStyles(ForwardRef(ListItem))>
`;

exports[`ListItemLink with isExternal specified true renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  color="textPrimary"
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  href="https://example.com/foo"
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <ArrowRightIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    inset={false}
    primary="Testing"
  />
</WithStyles(ForwardRef(ListItem))>
`;
